### PR TITLE
[Java][Client]com.github.curious-odd-man.rgxgen version update

### PR DESCRIPTION
**Pull request by code-lucidal58**
Fix Rxgen version update 

Pattern string `^[a-zA-Z0-9-._:]*$` could not be processed in the 1.1 version of library. The issue was fixed here: https://github.com/curious-odd-man/RgxGen/issues/53

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) beforehand.
- [x] Run the shell script `./bin/generate-samples.sh`to update all Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. These must match the expectations made by your contribution. You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

@bbdouglas (2017/07) @sreeshas (2017/08) @jfiala (2017/08) @lukoyanov (2017/09) @cbornet (2017/09) @jeff9finger (2018/01) @karismann (2019/03) @Zomzog (2019/04) @lwlee2608 (2019/10) @bkabrda (2020/01)

### CODE REVIEW

File: modules/openapi-generator/pom.xml
Comment (wing328) [inline modules/openapi-generator/pom.xml:261]: Don't think we can use SNAPSHOT version in the upcoming v5.0.0 major release.

Comment (code-lucidal58) [reply modules/openapi-generator/pom.xml:261]: In that case, shall I delete this PR. Because 1.3 is not yet released.

Comment (wing328) [reply modules/openapi-generator/pom.xml:261]: What about upgrading to 1.2? Would that address the issue you reported?

Comment (code-lucidal58) [reply modules/openapi-generator/pom.xml:261]: No

Comment (wing328) [reply modules/openapi-generator/pom.xml:261]: Please still upgrade it to 1.2 for the time being before the 1.3 release.

Comment (code-lucidal58) [reply modules/openapi-generator/pom.xml]: done

Comment (spacether) [reply modules/openapi-generator/pom.xml]: The maintainer of Rxgen may be able to put out a non-snapshot release before our major release. Should we wait until ~3 days before our release to see if he releases?

Comment (wing328) [reply modules/openapi-generator/pom.xml]: Sure we can wait.

Comment (curious-odd-man) [reply modules/openapi-generator/pom.xml]: Hello @spacether, @wing328 and @code-lucidal58.

The RgxGen version 1.3 is released and available in maven central.

**Comment by code-lucidal58** [comment]
New error encountered: a testcase failed
```
[ERROR] Tests run: 1341, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 38.199 s <<< FAILURE! - in TestSuite
[ERROR] testRegularExpressionOpenAPISchemaVersion3(org.openapitools.codegen.python.PythonLegacyClientCodegenTest)  Time elapsed: 0.043 s  <<< FAILURE!
com.github.curiousoddman.rgxgen.parsing.dflt.PatternDoesNotMatchAnythingException:
Before caret only new line is allowed!
'/^patt'
  ^
	at org.openapitools.codegen.python.PythonLegacyClientCodegenTest.testRegularExpressionOpenAPISchemaVersion3(PythonLegacyClientCodegenTest.java:81)

[INFO]
[INFO] Results:
[INFO]
[ERROR] Failures:
[ERROR]   PythonLegacyClientCodegenTest.testRegularExpressionOpenAPISchemaVersion3:81 » PatternDoesNotMatchAnything

**Comment by spacether** [comment]
> New error encountered: a testcase failed
> 
> ```
> [ERROR] Tests run: 1341, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 38.199 s <<< FAILURE! - in TestSuite
> [ERROR] testRegularExpressionOpenAPISchemaVersion3(org.openapitools.codegen.python.PythonLegacyClientCodegenTest)  Time elapsed: 0.043 s  <<< FAILURE!
> com.github.curiousoddman.rgxgen.parsing.dflt.PatternDoesNotMatchAnythingException:
> Before caret only new line is allowed!
> '/^patt'
>   ^
> 	at org.openapitools.codegen.python.PythonLegacyClientCodegenTest.testRegularExpressionOpenAPISchemaVersion3(PythonLegacyClientCodegenTest.java:81)
> 
> [INFO]
> [INFO] Results:
> [INFO]
> [ERROR] Failures:
> [ERROR]   PythonLegacyClientCodegenTest.testRegularExpressionOpenAPISchemaVersion3:81 » PatternDoesNotMatchAnything
> ```

Can you strip any leading and trailing / and / from the regex pattern before it is passed into rxgen? That will solve this problem.

File: modules/openapi-generator/pom.xml
Comment (spacether) [reply modules/openapi-generator/pom.xml]: Thank you!

**Review by spacether — APPROVED** [review]
Thank you for your PR! This looks great!

**Comment by curious-odd-man** [comment]
Hello @spacether and @code-lucidal58.

I've got a feeling that the fix  is incorrect:

I think that instead of:
```
while (pattern.startsWith("/")) pattern = pattern.substring(0, pattern.length()-1);
while (pattern.endsWith("/")) pattern = pattern.substring(1);
```

there should be:

```
while (pattern.endsWith("/")) pattern = pattern.substring(0, pattern.length()-1);
while (pattern.startsWith("/")) pattern = pattern.substring(1);
```

i.e. when ends with slash - truncate from end, when starts with slash - truncate from start.

image

**Comment by code-lucidal58** [comment]
Pardon me for this silly mistake. Here is the edit: https://github.com/OpenAPITools/openapi-generator/pull/8219

**Pull request merged**
